net/instaweb/rewriter/rewrite_driver.cc

```diff
@@ -2873,6 +2873,7 @@ bool RewriteDriver::InitiateRewrite(RewriteContext* rewrite_context) {
   // us with a partially constructed slot-graph.
   if (!can_rewrite_resources_) {
     if (rewrites_.empty()) {
+      srcset_collections_.clear();
```

Comment:
first of all, great catch.  I think Maks needs to weigh in on if this is the right thing.  If it is, I wonder if we should factor out a ClearSlots() helper that has:

    slots_.clear();
    inline_slots_.clear();
    inline_attribute_slots_.clear();
    srcset_collections_.clear();

and call that both here and on line 816.

Reply:
I agree that would make sense .. so if Maks blesses this change I'll factor out `RewriteDriver::ClearSlots()`